### PR TITLE
docs(redteam): add documentation for Custom and PII plugins

### PR DESCRIPTION
**Pull request by mldangelo**
- Create custom.md to document the Custom Plugin
- Create pii.md to document the PII Plugin
- Add pluginDocumentation.test.ts to ensure documentation coverage

### CODE REVIEW

File: site/docs/red-team/plugins/custom.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+```yaml
+redteam:
+  plugins:
+    - id: 'custom'
+      filePath: './path/to/your/custom-plugin.json'
+```
+
+The custom plugin JSON file should have the following structure:
+
+```json
```
Comment (typpo) [inline site/docs/red-team/plugins/custom.md]: this should be yaml

File: site/docs/red-team/plugins/custom.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  "generator": "Your custom generator logic here",
+  "grader": "Your custom grader logic here"
+}
+```
+
+## How It Works
+
+The Custom Plugin loads the plugin definition from the specified JSON file and uses it to generate test cases and grade responses.
+
+1. Generator: The `generator` field in your JSON file should contain a string template that defines how to generate test prompts.
```
Comment (typpo) [inline site/docs/red-team/plugins/custom.md]: yaml

File: site/docs/red-team/plugins/custom.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+2. Grader: The `grader` field should contain a string template that defines how to evaluate the AI's responses.
+
+Both the generator and grader can use Nunjucks templating for dynamic content.
```
Comment (typpo) [inline site/docs/red-team/plugins/custom.md]: what vars are supported?

File: site/docs/red-team/plugins/custom.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+## Implementation Details
```
Comment (typpo) [inline site/docs/red-team/plugins/custom.md]: not needed

File: site/docs/red-team/plugins/custom.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+```11:15:promptfoo/src/redteam/plugins/custom.ts
+  .object({
+    generator: z.string().min(1, 'Generator must not be empty').trim(),
+    grader: z.string().min(1, 'Grader must not be empty').trim(),
+  })
+  .strict();
+```
+
+## Best Practices
```
Comment (typpo) [inline site/docs/red-team/plugins/custom.md]: not useful

File: site/docs/red-team/plugins/custom.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+1. Thoroughly test your custom generator and grader logic before deployment.
+2. Use clear and descriptive names for your custom plugin files.
+3. Document the purpose and expected behavior of your custom plugins.
+4. Regularly review and update custom plugins to ensure they remain relevant and effective.
+
+## Limitations
```
Comment (typpo) [inline site/docs/red-team/plugins/custom.md]: probably not useful either

File: site/docs/red-team/plugins/custom.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+- The effectiveness of the Custom Plugin depends on the quality and comprehensiveness of your custom logic.
+- Complex custom logic may increase processing time for red teaming tests.
+- Ensure that your custom logic doesn't introduce unintended biases or vulnerabilities.
+
+## Importance in Gen AI Red Teaming
```
Comment (typpo) [inline site/docs/red-team/plugins/custom.md]: not really applicable

File: site/docs/red-team/plugins/pii.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+```
+
+## Subcategories
+
+The PII plugin includes the following subcategories:
+
+| Subcategory           | Plugin ID   | Description                                                                                     |
+| --------------------- | ----------- | ----------------------------------------------------------------------------------------------- |
+| Direct Disclosure     | pii:direct  | Tests if the model directly reveals PII when explicitly asked                                   |
+| Cross-Session Leakage | pii:session | Checks if the model leaks PII across different user sessions or interactions                    |
```
Comment (typpo) [inline site/docs/red-team/plugins/pii.md]: wonder if we should remove this now that we have a real `cross-session-leak` plugin

File: site/docs/red-team/plugins/pii.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+| API/Database Access   | pii:api-db  | Ensures the model doesn't attempt to access APIs or databases to retrieve PII                   |
+
+## How It Works
+
+The PII plugin generates test cases for each subcategory using templates and examples. These test cases are designed to challenge the AI's ability to protect personal information across various scenarios.
+
+```typescript:promptfoo/src/redteam/plugins/pii.ts
+startLine: 23
+endLine: 102
+```
```
Comment (typpo) [inline site/docs/red-team/plugins/pii.md]: ?

File: site/docs/red-team/plugins/pii.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+   ```
+
+## Evaluation Criteria
+
+The plugin assesses the AI's response based on several factors:
+
+```typescript:promptfoo/src/redteam/plugins/pii.ts
+startLine: 207
+endLine: 215
+```
```
Comment (typpo) [inline site/docs/red-team/plugins/pii.md]: :-1:

File: site/docs/red-team/plugins/pii.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+## Implementation Details
```
Comment (typpo) [inline site/docs/red-team/plugins/pii.md]: not needed

File: site/docs/red-team/plugins/pii.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+
+The PII plugin uses a grading system to evaluate the AI's responses. It employs a rubric that considers the system's purpose, user input, and allowed entities. The grader checks for various types of PII disclosure and determines whether the response passes or fails the PII disclosure check.
+
+```typescript:promptfoo/src/redteam/plugins/pii.ts
+startLine: 183
+endLine: 235
+```
+
+## Best Practices
```
Comment (typpo) [inline site/docs/red-team/plugins/pii.md]: not useful

**Review by typpo — CHANGES_REQUESTED** [review]
I'm glad AI is not taking our jobs yet

**Review by vsauter — APPROVED** [review]
LGTM DIDN'T READ

**Pull request merged**